schema/version.go

```diff
@@ -0,0 +1,36 @@
+package schema
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/Masterminds/semver"
+)
+
+// Version represents the schema version of an object
+type Version string
+
+// Validate the provided schema version is present and adheres
+// to semantic versioning
+func (v Version) Validate() error {
+	version := string(v)
+
+	_, err := semver.NewVersion(version)
+	if err != nil {
+		return fmt.Errorf("invalid schema version %q: %v", version, err)
+	}
+	return nil
+}
+
+// GetSemver returns a proper semver Version from the provided string,
+// trimming the non-semver prefix according to schema versioning protocol in the
+// cnabio/cnab-spec repo
+func GetSemver(schemaVersion string) (Version, error) {
```

Comment:
Would you please add a test for this function?

Reply:
Added, which uncovered a bit of lacking functionality in the method under test -- thank you!